src/Main/DetectiveGameMain.java

```diff
@@ -0,0 +1,132 @@
+package Main;
+
+import Commands.Command;
+import Commands.CommandFactory;
+import Core.*;
+import Extractors.*;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+//for New Branch
+
+public class DetectiveGameMain {
+    public static void main(String[] args) {
+        List<CaseFile> cases = loadCases();
+        displayCaseMenu(cases);
+        CaseFile selectedCase = selectCase(cases);
+        startGame(selectedCase);
+    }
+
+    private static List<CaseFile> loadCases() {
+        List<CaseFile> cases = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        File folder = new File("cases");
+
+        if (folder.exists()) {
+            File[] files = folder.listFiles((dir, name) -> name.endsWith(".json"));
+            if (files != null) {
+                for (File file : files) {
+                    try {
+                        CaseFile caseFile = mapper.readValue(file, CaseFile.class);
+                        cases.add(caseFile);
+                    } catch (Exception e) {
+                        System.out.println("Error loading case: " + file.getName());
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+        return cases;
+    }
+
+    private static void displayCaseMenu(List<CaseFile> cases) {
+        System.out.println("╔══════════════════════════════════════════════════════════════════════════════════════════════╗");
+        System.out.println("║                                     SELECT A CASE TO INVESTIGATE                             ║");
+        System.out.println("╠══════════════════════════════════════════════════════════════════════════════════════════════╣");
+
+        if (cases.isEmpty()) {
+            System.out.println("║ No cases available. Please add cases to the 'cases' folder.                                    ║");
+        } else {
+            for (int i = 0; i < cases.size(); i++) {
+                System.out.printf("║ %d. %-85s ║%n", i + 1, cases.get(i).getTitle());
+            }
+        }
+
+        System.out.println("╚══════════════════════════════════════════════════════════════════════════════════════════════╝");
+    }
+
+    private static CaseFile selectCase(List<CaseFile> cases) {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
Scanners should be closed and not initialized everytime you need to use it

See try with resources in java